spec/my_each_spec.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require_relative '../lib/my_enumerables'
+
+RSpec.describe Array do
+  subject(:array) { [1, 1, 2, 3, 5, 8, 13, 21, 34] }
+
+  describe '#my_each' do
+    it 'matches #each' do
```

Comment:
I like it! One thing though is that each only returns an array when given a block and an enumerable otherwise. I'm going to put this in a context block to make it clear.

For the `executes the block for each element` spec, what do you think about just repeating the same operations for `each` and `my_each` instead of using `map`, for clarity?

```ruby
it 'executes the block for each element' do
  my_each_results = []
  each_results = []

  array.my_each do |element|
    my_each_results << element * 2
  end

  array.each do |element|
    each_results << element * 2
  end

  expect(my_each_results).to eq(each_results)
end
```

Reply:
Sorry it has taken me so long to get back to you on this @rankoliang 😞 

Yeah, that looks way better than using map 💯 

Just to keep things simple I don't think we should worry about testing they return enumerators without the block. For now, at least, we could maybe add that in a future iteration.